signal SIGSEGV: segmentation violation on helm-service

<!--
For Security vulnerabilities, please refer to our security page https://github.com/keptn/keptn/blob/master/SECURITY.md
-->

**Environment**
<!-- Please tell us about the environment in which this bug can be reproduced in -->

* Client OS (e.g., Linux, macOS, Windows): Linux
* Keptn Version (`keptn version`): 0.13.1
* Kubernetes Cloud Provider (e.g., GKE, AKS): OpenShift
* Kubernetes version (`kubectl version`): 
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:59:11Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3+b93fd35", GitCommit:"3a0f2c90b43e6cffd07f57b5b78dd9f083e47ee2", GitTreeState:"clean", BuildDate:"2022-02-11T05:26:59Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}

<!-- Please also let us know about other components and their version if you believe that this issue is related to them (e.g., Istio, Web Browser Version). -->

**Affected Component**
<!-- Please tell us which component of Keptn is affected (leave empty if you are unsure). -->

* [ ] Approval
* [ ] Bridge
* [ ] CLI
* [ ] Datastore
* [ ] Docs
* [ ] Distributor / Event Broker
* [ ] Helm
* [ ] Istio
* [ ] Jmeter
* [ ] Kubernetes Integration
* [ ] Openshift Integration
* [ ] REST API
* [ ] Webhook

**Describe the bug**
<!-- Please provide a clear and concise description of what the bug is. Code samples should be put in the **To Reproduce** section. -->
Once helm-service is installed in a multicluster setup, after releasing the chart it always restarts because of this error:

> {"timestamp":"2022-03-03T15:18:25.775564696Z","logLevel":"INFO","message":"Starting deployment for service test-20220303-5-svc in stage svil of project test-20220303-5"}
> {"timestamp":"2022-03-03T15:18:25.784516125Z","logLevel":"INFO","message":"Updating values for service test-20220303-5-svc in stage svil of project test-20220303-5"}
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x25a83c5]
> goroutine 2495 [running]:
> github.com/keptn/keptn/helm-service/pkg/configurationchanger.(*ValuesManipulator).Manipulate(0xc000506348, 0x0, 0x13, 0x0)
>     /go/src/github.com/keptn/keptn/helm-service/pkg/configurationchanger/values_manipulator.go:25 +0xc5
> github.com/keptn/keptn/helm-service/pkg/configurationchanger.(*ConfigurationChanger).UpdateLoadedChart(0xc00023cc68, 0x0, 0xc000416f80, 0xf, 0xc000416f7c, 0x4, 0xc000058990, 0x13, 0x0, 0x0, ...)
>     /go/src/github.com/keptn/keptn/helm-service/pkg/configurationchanger/configuration_changer.go:34 +0xca
> github.com/keptn/keptn/helm-service/controller.(*DeploymentHandler).HandleEvent(0xc000621600, 0x2da30c0, 0xc000264d20, 0xc0000dc000, 0x173, 0x180, 0x0, 0x0)
>     /go/src/github.com/keptn/keptn/helm-service/controller/deployment_handler.go:67 +0xdc5
> main.gracefulMiddleware(0x2d786e0, 0xc0001fa000, 0x2da30c0, 0xc000264d20, 0xc0000dc000, 0x173, 0x180, 0x0, 0x0, 0x2d97448, ...)
>     /go/src/github.com/keptn/keptn/helm-service/main.go:243 +0xd1
> created by main.gotEvent
>     /go/src/github.com/keptn/keptn/helm-service/main.go:102 +0xfdc
> goroutine 1 [select, 25 minutes]:
> github.com/cloudevents/sdk-go/v2/protocol/http.(*Protocol).OpenInbound(0xc00011ed00, 0x2d78440, 0xc0001fe2c0, 0x0, 0x0)
>     /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/protocol/http/protocol_lifecycle.go:57 +0x365
> github.com/cloudevents/sdk-go/v2/client.(*ceClient).StartReceiver(0xc00011f040, 0x2d78440, 0xc0001fe280, 0x2786aa0, 0x2b80a08, 0x0, 0x0)
>     /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/client/client.go:265 +0x458
> main._main(0xc000118200, 0x0, 0x0, 0x1f90, 0x264bf56, 0x1, 0xc00005401a, 0x4, 0xc000000180)
>     /go/src/github.com/keptn/keptn/helm-service/main.go:200 +0x388
> main.main()
>     /go/src/github.com/keptn/keptn/helm-service/main.go:62 +0x199
> goroutine 19 [chan receive]:
> k8s.io/klog/v2.(*loggingT).flushDaemon(0x3bebb00)
>     /go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:1169 +0x8b
> created by k8s.io/klog/v2.init.0
>     /go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:420 +0xdf

After the restard, the helm-service doesn't handle anymore other sequences.

**To Reproduce**

Steps to reproduce the behavior:
1. Install Keptn 0.13.1 via helm chart 
2. Install helm-service 0.13.1 via helm chart
3. Trigger sequence delivery

**Expected behavior**
<!--- Please provide a clear and concise description of what you expected to happen -->
helm-service correctly releasing helm-charts

**Current behavior**
<!--- Please tell us what happens instead of the expected behavior -->
After the first release, the pod restarts and doesn't release any more charts.

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->

**Additional context**
<!-- Add any other context about the problem here. -->

Hi @odubajDT , this is the git upstream repo where you can find everything: https://github.com/krateoplatformops/test-20220303-5-keptn

Basically what we are trying to do is to trigger the sequence with the following payload:

```
{
  "data": {
    "configurationChange": {
      "values": {
        "image": "image-registry.openshift-image-registry.svc:5000/cloudbees/test-20220303-5:latest"
      }
    },
    "deployment": {
      "deploymentURIsLocal": null,
      "deploymentstrategy": ""
    },
    "project": "test-20220303-5",
    "service": "test-20220303-5-svc",
    "stage": "svil"
  },
  "id": "45475a6a-9827-47b9-a658-8cee8a47a6c2",
  "shkeptncontext": "99ad9f95-848f-4c59-a781-31f78755658e",
  "shkeptnspecversion": "0.2.4",
  "source": "https://github.com/keptn/keptn/cli#configuration-change",
  "specversion": "1.0",
  "time": "2022-03-04T11:38:01.860Z",
  "type": "sh.keptn.event.svil.delivery-direct.triggered"
}
```

This is the log we found in the killed pod:

```
{"timestamp":"2022-03-04T13:12:10.672707506Z","logLevel":"INFO","message":"Starting deployment for service test-20220303-5-svc in stage svil of project test-20220303-5"}
{"timestamp":"2022-03-04T13:12:10.691889171Z","logLevel":"INFO","message":"Updating values for service test-20220303-5-svc in stage svil of project test-20220303-5"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x25a83c5]
goroutine 9068 [running]:
github.com/keptn/keptn/helm-service/pkg/configurationchanger.(*ValuesManipulator).Manipulate(0xc000010280, 0x0, 0x13, 0x0)
    /go/src/github.com/keptn/keptn/helm-service/pkg/configurationchanger/values_manipulator.go:25 +0xc5
github.com/keptn/keptn/helm-service/pkg/configurationchanger.(*ConfigurationChanger).UpdateLoadedChart(0xc00020ac68, 0x0, 0xc000558bc0, 0xf, 0xc000558ba8, 0x4, 0xc0006c8480, 0x13, 0x0, 0x0, ...)
    /go/src/github.com/keptn/keptn/helm-service/pkg/configurationchanger/configuration_changer.go:34 +0xca
github.com/keptn/keptn/helm-service/controller.(*DeploymentHandler).HandleEvent(0xc00007f700, 0x2da30c0, 0xc0000209a0, 0xc000024000, 0x198, 0x1a0, 0x0, 0x0)
    /go/src/github.com/keptn/keptn/helm-service/controller/deployment_handler.go:67 +0xdc5
main.gracefulMiddleware(0x2d786e0, 0xc00001e020, 0x2da30c0, 0xc0000209a0, 0xc000024000, 0x198, 0x1a0, 0x0, 0x0, 0x2d97448, ...)
    /go/src/github.com/keptn/keptn/helm-service/main.go:243 +0xd1
created by main.gotEvent
    /go/src/github.com/keptn/keptn/helm-service/main.go:102 +0xfdc
goroutine 1 [select, 94 minutes]:
github.com/cloudevents/sdk-go/v2/protocol/http.(*Protocol).OpenInbound(0xc00056c270, 0x2d78440, 0xc00079a2c0, 0x0, 0x0)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/protocol/http/protocol_lifecycle.go:57 +0x365
github.com/cloudevents/sdk-go/v2/client.(*ceClient).StartReceiver(0xc00056c410, 0x2d78440, 0xc00079a280, 0x2786aa0, 0x2b80a08, 0x0, 0x0)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/client/client.go:265 +0x458
main._main(0xc00004e220, 0x0, 0x0, 0x1f90, 0x264bf56, 0x1, 0xc00005401a, 0x4, 0xc000000180)
    /go/src/github.com/keptn/keptn/helm-service/main.go:200 +0x388
main.main()
    /go/src/github.com/keptn/keptn/helm-service/main.go:62 +0x199
goroutine 6 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x3bebb00)
    /go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:1169 +0x8b
created by k8s.io/klog/v2.init.0
    /go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:420 +0xdf
goroutine 40 [syscall, 94 minutes]:
os/signal.signal_recv(0x2b7ea70)
    /usr/local/go/src/runtime/sigqueue.go:168 +0xa5
os/signal.loop()
    /usr/local/go/src/os/signal/signal_unix.go:23 +0x25
created by os/signal.Notify.func1.1
    /usr/local/go/src/os/signal/signal.go:151 +0x45
goroutine 41 [chan receive, 94 minutes]:
main.getGracefulContext.func1(0xc000554ba0, 0xc000558000, 0xc000378ad0)
    /go/src/github.com/keptn/keptn/helm-service/main.go:229 +0x38
created by main.getGracefulContext
    /go/src/github.com/keptn/keptn/helm-service/main.go:228 +0x18f
goroutine 42 [select]:
github.com/cloudevents/sdk-go/v2/protocol/http.(*Protocol).Respond(0xc00056c270, 0x2d78440, 0xc00079a2c0, 0xc000378b10, 0x2d78718, 0xc0006d01e0, 0xc0002a7230, 0xc000558030)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/protocol/http/protocol.go:267 +0xd8
github.com/cloudevents/sdk-go/v2/client.(*ceClient).StartReceiver.func2(0xc000558030, 0xc00056c410, 0xc000378b10)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/client/client.go:236 +0x302
created by github.com/cloudevents/sdk-go/v2/client.(*ceClient).StartReceiver
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/client/client.go:228 +0x3a5
goroutine 43 [select, 94 minutes]:
github.com/cloudevents/sdk-go/v2/protocol/http.(*Protocol).Respond(0xc00056c270, 0x2d78440, 0xc00079a2c0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/protocol/http/protocol.go:267 +0xd8
github.com/cloudevents/sdk-go/v2/client.(*ceClient).StartReceiver.func2(0xc000558030, 0xc00056c410, 0xc000378b10)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/client/client.go:236 +0x302
created by github.com/cloudevents/sdk-go/v2/client.(*ceClient).StartReceiver
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/client/client.go:228 +0x3a5
goroutine 44 [select, 94 minutes]:
github.com/cloudevents/sdk-go/v2/protocol/http.(*Protocol).Respond(0xc00056c270, 0x2d78440, 0xc00079a2c0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/protocol/http/protocol.go:267 +0xd8
github.com/cloudevents/sdk-go/v2/client.(*ceClient).StartReceiver.func2(0xc000558030, 0xc00056c410, 0xc000378b10)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/client/client.go:236 +0x302
created by github.com/cloudevents/sdk-go/v2/client.(*ceClient).StartReceiver
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/client/client.go:228 +0x3a5
goroutine 45 [select, 94 minutes]:
github.com/cloudevents/sdk-go/v2/protocol/http.(*Protocol).Respond(0xc00056c270, 0x2d78440, 0xc00079a2c0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/protocol/http/protocol.go:267 +0xd8
github.com/cloudevents/sdk-go/v2/client.(*ceClient).StartReceiver.func2(0xc000558030, 0xc00056c410, 0xc000378b10)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/client/client.go:236 +0x302
created by github.com/cloudevents/sdk-go/v2/client.(*ceClient).StartReceiver
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/client/client.go:228 +0x3a5
goroutine 46 [IO wait]:
internal/poll.runtime_pollWait(0x7f7fee6c3eb8, 0x72, 0x0)
    /usr/local/go/src/runtime/netpoll.go:227 +0x55
internal/poll.(*pollDesc).wait(0xc00079d498, 0x72, 0x0, 0x0, 0x2a94895)
    /usr/local/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
    /usr/local/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Accept(0xc00079d480, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/go/src/internal/poll/fd_unix.go:401 +0x212
net.(*netFD).accept(0xc00079d480, 0xd68845bfb0d9d2e4, 0x0, 0x0)
    /usr/local/go/src/net/fd_unix.go:172 +0x45
net.(*TCPListener).accept(0xc00000e078, 0x6222102a, 0xc00015de10, 0xdb0b86)
    /usr/local/go/src/net/tcpsock_posix.go:139 +0x32
net.(*TCPListener).Accept(0xc00000e078, 0xc00015de60, 0x18, 0xc000552a80, 0x105af9b)
    /usr/local/go/src/net/tcpsock.go:261 +0x65
net/http.(*Server).Serve(0xc00031a000, 0x2d72f90, 0xc00000e078, 0x0, 0x0)
    /usr/local/go/src/net/http/server.go:2962 +0x285
github.com/cloudevents/sdk-go/v2/protocol/http.(*Protocol).OpenInbound.func2(0xc0006182a0, 0xc00056c270, 0x2d72f90, 0xc00000e078)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/protocol/http/protocol_lifecycle.go:53 +0x46
created by github.com/cloudevents/sdk-go/v2/protocol/http.(*Protocol).OpenInbound
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/protocol/http/protocol_lifecycle.go:52 +0x2d1
goroutine 9054 [IO wait]:
internal/poll.runtime_pollWait(0x7f7fee6c3dd0, 0x72, 0xffffffffffffffff)
    /usr/local/go/src/runtime/netpoll.go:227 +0x55
internal/poll.(*pollDesc).wait(0xc0004a0e98, 0x72, 0x1000, 0x1000, 0xffffffffffffffff)
    /usr/local/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
    /usr/local/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc0004a0e80, 0xc0001fd000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/go/src/internal/poll/fd_unix.go:166 +0x1d5
net.(*netFD).Read(0xc0004a0e80, 0xc0001fd000, 0x1000, 0x1000, 0xc0001f6e88, 0xc0001f6d18, 0x2)
    /usr/local/go/src/net/fd_posix.go:55 +0x4f
net.(*conn).Read(0xc0005b01e8, 0xc0001fd000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/go/src/net/net.go:183 +0x91
net/http.(*persistConn).Read(0xc0005c58c0, 0xc0001fd000, 0x1000, 0x1000, 0xc0001f6dc4, 0x2, 0x2)
    /usr/local/go/src/net/http/transport.go:1922 +0x77
bufio.(*Reader).fill(0xc000554720)
    /usr/local/go/src/bufio/bufio.go:101 +0x108
bufio.(*Reader).Peek(0xc000554720, 0x1, 0x0, 0x1, 0x1, 0x1, 0x0)
    /usr/local/go/src/bufio/bufio.go:139 +0x4f
net/http.(*persistConn).readLoop(0xc0005c58c0)
    /usr/local/go/src/net/http/transport.go:2083 +0x1a8
created by net/http.(*Transport).dialConn
    /usr/local/go/src/net/http/transport.go:1743 +0xc77
goroutine 9065 [IO wait]:
internal/poll.runtime_pollWait(0x7f7fee6c3b18, 0x72, 0xffffffffffffffff)
    /usr/local/go/src/runtime/netpoll.go:227 +0x55
internal/poll.(*pollDesc).wait(0xc000796798, 0x72, 0x1000, 0x1000, 0xffffffffffffffff)
    /usr/local/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
    /usr/local/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc000796780, 0xc00016d000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/go/src/internal/poll/fd_unix.go:166 +0x1d5
net.(*netFD).Read(0xc000796780, 0xc00016d000, 0x1000, 0x1000, 0x0, 0x7f7fee6c3b20, 0x0)
    /usr/local/go/src/net/fd_posix.go:55 +0x4f
net.(*conn).Read(0xc000010110, 0xc00016d000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/go/src/net/net.go:183 +0x91
net/http.(*connReader).Read(0xc0002a7170, 0xc00016d000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/go/src/net/http/server.go:780 +0x1b9
bufio.(*Reader).fill(0xc000499c20)
    /usr/local/go/src/bufio/bufio.go:101 +0x108
bufio.(*Reader).Peek(0xc000499c20, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/go/src/bufio/bufio.go:139 +0x4f
net/http.(*conn).readRequest(0xc000419ea0, 0x2d78440, 0xc00007f400, 0x0, 0x0, 0x0)
    /usr/local/go/src/net/http/server.go:963 +0xec5
net/http.(*conn).serve(0xc000419ea0, 0x2d784e8, 0xc00007f400)
    /usr/local/go/src/net/http/server.go:1859 +0x705
created by net/http.(*Server).Serve
    /usr/local/go/src/net/http/server.go:2994 +0x39b
goroutine 9075 [select]:
net/http.(*persistConn).writeLoop(0xc00048f0e0)
    /usr/local/go/src/net/http/transport.go:2382 +0xf7
created by net/http.(*Transport).dialConn
    /usr/local/go/src/net/http/transport.go:1744 +0xc9c
goroutine 9074 [IO wait]:
internal/poll.runtime_pollWait(0x7f7fee6c3c00, 0x72, 0xffffffffffffffff)
    /usr/local/go/src/runtime/netpoll.go:227 +0x55
internal/poll.(*pollDesc).wait(0xc0004a0f18, 0x72, 0x1000, 0x1000, 0xffffffffffffffff)
    /usr/local/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
    /usr/local/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc0004a0f00, 0xc00026f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/go/src/internal/poll/fd_unix.go:166 +0x1d5
net.(*netFD).Read(0xc0004a0f00, 0xc00026f000, 0x1000, 0x1000, 0xd5699c, 0xc000159c38, 0xd82260)
    /usr/local/go/src/net/fd_posix.go:55 +0x4f
net.(*conn).Read(0xc0005b0208, 0xc00026f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/go/src/net/net.go:183 +0x91
net/http.(*persistConn).Read(0xc00048f0e0, 0xc00026f000, 0x1000, 0x1000, 0xc0000a55c0, 0xc000159d40, 0xd21eb5)
    /usr/local/go/src/net/http/transport.go:1922 +0x77
bufio.(*Reader).fill(0xc0005547e0)
    /usr/local/go/src/bufio/bufio.go:101 +0x108
bufio.(*Reader).Peek(0xc0005547e0, 0x1, 0x0, 0x1, 0x4, 0x1, 0x3)
    /usr/local/go/src/bufio/bufio.go:139 +0x4f
net/http.(*persistConn).readLoop(0xc00048f0e0)
    /usr/local/go/src/net/http/transport.go:2083 +0x1a8
created by net/http.(*Transport).dialConn
    /usr/local/go/src/net/http/transport.go:1743 +0xc77
goroutine 9055 [select]:
net/http.(*persistConn).writeLoop(0xc0005c58c0)
    /usr/local/go/src/net/http/transport.go:2382 +0xf7
created by net/http.(*Transport).dialConn
    /usr/local/go/src/net/http/transport.go:1744 +0xc9c
```

Helm-service is at version 0.13.1.

This is the pod log:

{"timestamp":"2022-03-08T10:44:42.143394395Z","logLevel":"INFO","message":"Starting deployment for service krateo-20220308 in stage svil of project krateo-20220308"}
{"timestamp":"2022-03-08T10:44:42.197114575Z","logLevel":"INFO","message":"Updating values for service krateo-20220308 in stage svil of project krateo-20220308"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x25a83c5]
goroutine 1284 [running]:
github.com/keptn/keptn/helm-service/pkg/configurationchanger.(*ValuesManipulator).Manipulate(0xc0005a6070, 0x0, 0xf, 0x0)
    /go/src/github.com/keptn/keptn/helm-service/pkg/configurationchanger/values_manipulator.go:25 +0xc5
github.com/keptn/keptn/helm-service/pkg/configurationchanger.(*ConfigurationChanger).UpdateLoadedChart(0xc000232c68, 0x0, 0xc0006a2ad0, 0xf, 0xc0006a2ab8, 0x4, 0xc0006a2ae0, 0xf, 0x0, 0x0, ...)
    /go/src/github.com/keptn/keptn/helm-service/pkg/configurationchanger/configuration_changer.go:34 +0xca
github.com/keptn/keptn/helm-service/controller.(*DeploymentHandler).HandleEvent(0xc000497500, 0x2da30c0, 0xc0002d2a80, 0xc0002e2000, 0x181, 0x1a0, 0x0, 0x0)
    /go/src/github.com/keptn/keptn/helm-service/controller/deployment_handler.go:67 +0xdc5
main.gracefulMiddleware(0x2d786e0, 0xc000830020, 0x2da30c0, 0xc0002d2a80, 0xc0002e2000, 0x181, 0x1a0, 0x0, 0x0, 0x2d97448, ...)
    /go/src/github.com/keptn/keptn/helm-service/main.go:243 +0xd1
created by main.gotEvent
    /go/src/github.com/keptn/keptn/helm-service/main.go:102 +0xfdc
goroutine 1 [select, 12 minutes]:
github.com/cloudevents/sdk-go/v2/protocol/http.(*Protocol).OpenInbound(0xc000145a00, 0x2d78440, 0xc000496e00, 0x0, 0x0)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/protocol/http/protocol_lifecycle.go:57 +0x365
github.com/cloudevents/sdk-go/v2/client.(*ceClient).StartReceiver(0xc000145ad0, 0x2d78440, 0xc000496dc0, 0x2786aa0, 0x2b80a08, 0x0, 0x0)
    /go/pkg/mod/github.com/cloudevents/sdk-go/v2@v2.8.0/client/client.go:265 +0x458
main._main(0xc00004e220, 0x0, 0x0, 0x1f90, 0x264bf56, 0x1, 0xc00005401a, 0x5, 0xc000000180)
    /go/src/github.com/keptn/keptn/helm-service/main.go:200 +0x388
main.main()
    /go/src/github.com/keptn/keptn/helm-service/main.go:62 +0x199
goroutine 6 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x3bebb00)
    /go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:1169 +0x8b
created by k8s.io/klog/v2.init.0
    /go/pkg/mod/k8s.io/klog/v2@v2.9.0/klog.go:420 +0xdf
goroutine 13 [chan receive, 12 minutes]:
main.getGracefulContext.

--- a/helm-service/controller/deployment_handler.go
+++ b/helm-service/controller/deployment_handler.go
@@ -3,6 +3,8 @@ package controller
 import (
 	"errors"
 	"fmt"
+	"math"
+
 	cloudevents "github.com/cloudevents/sdk-go/v2"
 	keptnevents "github.com/keptn/go-utils/pkg/lib"
 	keptnv2 "github.com/keptn/go-utils/pkg/lib/v0_2_0"
@@ -11,7 +13,6 @@ import (
 	"github.com/keptn/keptn/helm-service/pkg/mesh"
 	"helm.sh/helm/v3/pkg/chart"
 	corev1 "k8s.io/api/core/v1"
-	"math"
 )
 
 // DeploymentHandler is a handler for doing the deployment and
@@ -63,11 +64,16 @@ func (h *DeploymentHandler) HandleEvent(ce cloudevents.Event) {
 		valuesUpdater := configurationchanger.NewValuesManipulator(e.ConfigurationChange.Values)
 
 		userChart, _, err = h.getUserChart(e.EventData, commitID)
+		if err != nil {
+			err = fmt.Errorf("failed to load chart: %w", err)
+			h.handleError(ce.ID(), err, keptnv2.DeploymentTaskName, h.getFinishedEventDataForError(e.EventData, err))
+			return
+		}
 		userChart, commitID, err = configurationchanger.NewConfigurationChanger(
 			h.getConfigServiceURL()).UpdateLoadedChart(userChart, e.EventData, false, valuesUpdater)
 
 		if err != nil {
-			err = fmt.Errorf("failed to update values: %v", err)
+			err = fmt.Errorf("failed to update values: %w", err)
 			h.handleError(ce.ID(), err, keptnv2.DeploymentTaskName, h.getFinishedEventDataForError(e.EventData, err))
 			return
 		}
@@ -75,7 +81,7 @@ func (h *DeploymentHandler) HandleEvent(ce cloudevents.Event) {
 		// Read chart
 		userChart, commitID, err = h.getUserChart(e.EventData, commitID)
 		if err != nil {
-			err = fmt.Errorf("failed to load chart: %v", err)
+			err = fmt.Errorf("failed to load chart: %w", err)
 			h.handleError(ce.ID(), err, keptnv2.DeploymentTaskName, h.getFinishedEventDataForError(e.EventData, err))
 			return
 		}

--- a/helm-service/controller/deployment_handler_test.go
+++ b/helm-service/controller/deployment_handler_test.go
@@ -346,6 +346,64 @@ func TestHandleEventWithoutDeploymentURLAndUserManagedDeploymentStrategy(t *test
 	assert.Equal(t, keptn.UserManaged, mockedBaseHandler.upgradeChartInvocations[0].strategy)
 }
 
+func TestHandleEventWithoutResources(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockedBaseHandler := NewMockedHandler(createKeptn(), "", func(options *MockedHandlerOptions) {
+		options.GetUserManagedEndpointsBehavior = func(event keptnv2.EventData) (*keptnv2.Endpoints, error) {
+			return nil, nil
+		}
+		options.GetUserChartsErrBehaviour = true
+
+	})
+
+	mockedOnboarder := mocks.NewMockOnboarder(ctrl)
+	mockedChartGenerator := mocks.NewMockChartGenerator(ctrl)
+
+	deploymentHandler := DeploymentHandler{
+		Handler:               mockedBaseHandler,
+		mesh:                  mocks.NewMockMesh(ctrl),
+		generatedChartHandler: mockedChartGenerator,
+		onboarder:             mockedOnboarder,
+	}
+
+	deploymentTriggeredEventData := keptnv2.DeploymentTriggeredEventData{
+		EventData: keptnv2.EventData{
+			Project: "my-project",
+			Stage:   "production",
+			Service: "my-service",
+		},
+		ConfigurationChange: keptnv2.ConfigurationChange{
+			Values: map[string]interface{}{
+				"image": "nexus-registry.apps.ocp4-test.it/krateo-20220308:latest",
+			},
+		},
+		Deployment: keptnv2.DeploymentTriggeredData{
+			DeploymentStrategy: keptn.UserManaged.String(),
+		},
+	}
+
+	ce := cloudevents.NewEvent()
+	_ = ce.SetData(cloudevents.ApplicationJSON, deploymentTriggeredEventData)
+	ce.SetExtension("gitcommitid", "silly-one")
+	deploymentHandler.HandleEvent(ce)
+
+	expectedDeploymentFinishedEvent := keptnv2.DeploymentFinishedEventData{
+		EventData: keptnv2.EventData{
+			Project: "my-project",
+			Stage:   "production",
+			Service: "my-service",
+			Status:  keptnv2.StatusErrored,
+			Result:  keptnv2.ResultFailed,
+			Message: "failed to load chart: some error",
+		},
+	}
+
+	assert.Equal(t, 1, len(mockedBaseHandler.handledErrorEvents))
+	assert.Equal(t, expectedDeploymentFinishedEvent, mockedBaseHandler.handledErrorEvents[0])
+}
+
 func TestHandleUnparsableDeploymentEvent(t *testing.T) {
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()

--- a/helm-service/controller/mock_handler.go
+++ b/helm-service/controller/mock_handler.go
@@ -3,6 +3,7 @@ package controller
 import (
 	"errors"
 	"fmt"
+
 	keptnevents "github.com/keptn/go-utils/pkg/lib"
 
 	cloudevents "github.com/cloudevents/sdk-go/v2"
@@ -31,6 +32,7 @@ type MockedHandlerOption func(*MockedHandlerOptions)
 type MockedHandlerOptions struct {
 	SendEventBehavior               func(eventType string) bool
 	GetUserManagedEndpointsBehavior func(event keptnv2.EventData) (*keptnv2.Endpoints, error)
+	GetUserChartsErrBehaviour       bool
 }
 
 func sendingEventSucceeds(eventType string) bool {
@@ -75,6 +77,9 @@ func (h *MockedHandler) getGeneratedChart(e keptnv2.EventData, commitID string) 
 }
 
 func (h *MockedHandler) getUserChart(e keptnv2.EventData, commitID string) (*chart.Chart, string, error) {
+	if h.options.GetUserChartsErrBehaviour == true {
+		return nil, "", errors.New("some error")
+	}
 	ch := helm.GetTestUserChart()
 	return &ch, commitID, nil
 }

